tmt/libraries/__init__.py

```diff
@@ -148,15 +148,17 @@ def already_fetched(lib: Dependency) -> bool:
                 source_location=source_location, target_location=target_location)
             gathered_libraries.append(library)
             imported_lib_ids.append(library.identifier)
-            assert parent is not None  # narrow type
-            if library.hostname == 'local':
-                library_path = library.fmf_node_path
-            else:  # TODO: Change with pruning for libraries
-                library_path = parent.workdir / library.dest / library.repo
 
             from .beakerlib import BeakerLib
             if isinstance(library, BeakerLib):
                 # Recursively check for possible dependent libraries
+                assert parent is not None  # narrow type
+                if library.hostname == 'local':
+                    library_path = library.path
```

Comment:
Using `library.path` instead of `library.fmf_nod_path` means that the `fmf_node_path()` property is not used anymore. However, it should handle the case when the `fmf` tree is nested deeper in the git repository. See also the [fmf identifier definition](https://fmf.readthedocs.io/en/latest/concept.html#identifiers).

https://github.com/teemtee/tmt/blob/dfdd4afa4f93893169cbc047e8dcb4ad82bcf688/tmt/libraries/beakerlib.py#L154-L158

However, currently it seems that the use case when both `url` and `path` are provided in the library reference is broken. I've filed it as #2149. So, all in all, I believe this is an acceptable hot fix for this.

@lukaszachy, @happz, what do you think?

Reply:
It looks like `fmf_node_path` is used only by libraries, no other dependency class uses it, maybe we could remove it completely?

Reply:
Yeah, it's not used, but IMO we should get this in to get passes on PRs and I will remove or change the method in the pruning patch.